src/mds/Server.cc

```diff
@@ -1321,6 +1321,9 @@ void Server::evict_cap_revoke_non_responders() {
 }
 
 void Server::handle_conf_change(const std::set<std::string>& changed) {
+  if (changed.count("mds_lock_cache")){
```

Comment:
This needs added to the tracked configs in `MDSRankDispatcher::get_tracked_conf_keys()`?

This config may be more appropriately named `mds_allow_async_dirops`.

Reply:
> This needs added to the tracked configs in `MDSRankDispatcher::get_tracked_conf_keys()`?
> 

Yeah, will fix it.

> This config may be more appropriately named `mds_allow_async_dirops`.

Sound better.

Reply:
@batrick BTW, then we should also stop delegating the inode# to kclient to comply to the `mds_allow_async_dirops`.  

IMO this should be fine, because at least in mds side we will have a simple way to disable the `async dirops` instead of doing this in kclient mount one by one.

What do you think ?

Reply:
Recently was reminded of `mds_client_delegate_inos_pct` which can be set to `0`, effectively also disabling async dirops.

It would be nice if we had a single config to control async dirops which is more clear (like a simple well-named bool). I still support having the new name. However, we shoudl observe that even without delegated inodes that the MDS may create lock caches. So perhaps this config should keep its current name, although "mds_allow_lock_caching" would be better.

Reply:
> Recently was reminded of `mds_client_delegate_inos_pct` which can be set to `0`, effectively also disabling async dirops.
> 
Yeah, correct. I remember this now.